Workarounds following gcc7 issue. Need to be evaluated with upstream.

  CC       src/sif.lo
/scratch/userland-gate/components/open-fabrics/libsif/libsif-1.1/src/sif.c: In function 'sif_init_context':
/scratch/userland-gate/components/open-fabrics/libsif/libsif-1.1/src/sif.c:229:28: error: '/device/driver/module/parameters' directive writing 32 bytes into a region of size between 1 and 256 [-Werror=format-overflow=]
     sprintf(param_path, "%s/device/driver/module/parameters", sdev->verbs_dev.device.ibdev_path);
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/scratch/userland-gate/components/open-fabrics/libsif/libsif-1.1/src/sif.c:229:5: note: 'sprintf' output between 33 and 288 bytes into a destination of size 256
     sprintf(param_path, "%s/device/driver/module/parameters", sdev->verbs_dev.device.ibdev_path);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors  

--- a/src/sif.c
+++ b/src/sif.c
@@ -171,7 +171,7 @@
     struct sif_get_context_resp  resp;
     struct sif_device *sdev = to_sdev(ibdev);
     char value[20];
-    char param_path[IBV_SYSFS_PATH_MAX];
+    char param_path[IBV_SYSFS_PATH_MAX+32];	// gcc7 issue: directive writing 32 bytes into a region of size between 1 and 256
 #if defined(__SVR4) && defined(__sun)
 /*
  * See CR 22214822,22245798:collect buffer mode disabled for solaris
